cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/MessagingMetricsCollectionService.java

```diff
@@ -161,6 +170,11 @@ void publish(MessagingService messagingService) throws IOException {
               done = true;
             }
           }
+        } catch (IOException ioe) {
+          LOG.error("Received exception {} " +
```

Comment:
move the original exception message to the end.

Also, instead of logging here, it would be better to throw a new IOException with this message and the original exception as the cause. Then the caller can log it (it does anyway, right?)